### ISSUE TITLE
Valid regexp reported as invalid

### ISSUE THREAD

**Issue by haximilian** [eslint/eslint#11564]
**Tell us about your environment**

* **ESLint Version:** v5.15.3
* **Node Version:** v11.12.20
* **npm Version:** v6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "root": true,
  "extends": "eslint:recommended",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "env": {
    "amd": true,
    "es6": true,
    "node": true
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Source Code:
```js
let FILTER_STRINGS = /(?<!\\)"/;
```

ESLint Command:
```bash
eslint --ext .js --fix .
```

**What did you expect to happen?**
An error about an unused variable, but other than that, absolutely nothing. (It is a [valid regexp](https://regex101.com/r/aT4y9H/2), after all.

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint **crashed**, claiming the Regex was invalid:
> error  Parsing error: Invalid regular expression: /(?<!\\)"/: Invalid group

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. Negative lookbehinds in regexes are an ES2018 feature. To enable support for them you should change `ecmaVersion: 6` in your config to `ecmaVersion: 2018`.

**Comment by haximilian** [comment]
@not-an-aardvark Oh! I hadn't realized that they weren't included in the original regex spec. Thank you for your help!

**Comment by platinumazure** [comment]
Closing as the question has been answered. Feel free to stop by our [Gitter chat](https://gitter.im/eslint/eslint) if you have any other questions!
